### PR TITLE
Update e2e run script

### PR DESCRIPTION
**Pull request by vdemeester**
Update the e2e run script to be almost the same as `hack/make/test-integration`. 

In order to use this e2e image to test docker api :

```bash
$ docker build -f Dockerfile.e2e -t moby-e2e .
# […]
# To run all tests
$ docker run -v /var/run/docker.sock:/var/run/docker.sock \
    -e DOCKER_API_VERSION=1.37 \
    moby-e2e
# [ … ]
# To run only /test/integration/container
$ docker run -v /var/run/docker.sock:/var/run/docker.sock \
    -e TEST_INTEGRATION_DIR=/tests/integration/container \
    -e DOCKER_API_VERSION=1.37 \
    moby-e2e
# [ … ]
```

PS: it will fail now, but should help us write better tests (i.e. put the correct condition on running tests)
PS: I need to add a step in the CI to build it on master (but I'm tempted to wait for `Jenkinsfile` for that :stuck_out_tongue:)

Also adds support for s390x from #35449 

cc @chris-crone 

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

### CODE REVIEW

File: hack/test/e2e-run.sh [hunk 1]
```diff
@@ -1,41 +1,72 @@
 #!/usr/bin/env bash
-set -e
+set -e -o pipefail
```
Comment (dnephin) [inline hack/test/e2e-run.sh]: Can we set `-u` as well?

File: hack/test/e2e-run.sh [hunk 2]
```diff
@@ -1,41 +1,72 @@
...
+
+integration_api_dirs=${TEST_INTEGRATION_DIR:-"$(
+	find ./integration -type d |
+	grep -vE '(^./integration($|/internal)|/testdata)')"}
 
 run_test_integration() {
-  run_test_integration_suites
-  run_test_integration_legacy_suites
+	[[ "$TESTFLAGS" != *-check.f* ]] && run_test_integration_suites
+	run_test_integration_legacy_suites
```
Comment (dnephin) [inline hack/test/e2e-run.sh]: hopefully we can find a way to share this without having to duplicate it once we do more work on #35937

**Review by dnephin — APPROVED** [review]
LGTM, just one small nit

**Review by thaJeztah — APPROVED** [review]
LGTM after nit is addressed

ping @tianon in case you want to have a look

**Comment by vdemeester** [comment]
powerpc builds is hitting `no space left on device` 😅 These errors are unrelated to this change, I'll go ahead and merge but we need to enhance those jobs …

**Pull request merged**